sdk/storage/azure-storage-file-share/tests/perfstress/_test_base.py

```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import uuid
+
+from azure_devtools.perfstress_tests import PerfStressTest
+
+from azure.core.exceptions import ResourceNotFoundError
+from azure.storage.fileshare import ShareServiceClient as SyncShareServiceClient
+from azure.storage.fileshare.aio import ShareServiceClient as AsyncShareServiceClient
+
+
+class _ServiceTest(PerfStressTest):
+    service_client = None
+    async_service_client = None
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+        if not connection_string:
+            raise Exception("Undefined environment variable AZURE_STORAGE_CONNECTION_STRING")
+
+        if not _ServiceTest.service_client or self.args.service_client_per_instance:
+            if self.args.max_range_size:
+                _ServiceTest.service_client = SyncShareServiceClient.from_connection_string(conn_str=connection_string, max_range_size=self.args.max_range_size)
+                _ServiceTest.async_service_client = AsyncShareServiceClient.from_connection_string(conn_str=connection_string, max_range_size=self.args.max_range_size)
+            else:
+                _ServiceTest.service_client = SyncShareServiceClient.from_connection_string(conn_str=connection_string)
+                _ServiceTest.async_service_client = AsyncShareServiceClient.from_connection_string(conn_str=connection_string)
+
+        self.service_client = _ServiceTest.service_client
+        self.async_service_client =_ServiceTest.async_service_client
+
+    async def close(self):
+        await self.async_service_client.close()
+        await super().close()
+
+    @staticmethod
+    def add_arguments(parser):
+        super(_ServiceTest, _ServiceTest).add_arguments(parser)
+        parser.add_argument('--max-range-size', nargs='?', type=int, help='Maximum size of file uploading in single HTTP PUT. Defaults to 4*1024*1024', default=4*1024*1024)
+        parser.add_argument('--service-client-per-instance', action='store_true', help='Create one ServiceClient per test instance.  Default is to share a single ServiceClient.', default=False)
```

Comment:
Why is there a parameter for this?  Do we want to test one shared ServiceClient vs one instance per test class?  Do we expect to see different perf?  What do we recommend customers do?

Reply:
Hah - this was here because you had it in the original blobs test prototypes - so I assumed you had a particular reason and kept it around. Though I can remove it :)

Reply:
I recall it was something we wanted to test at one point.  Maybe related to connection sharing?  Let's keep this enabled for now and get results both ways and compare them.

Reply:
Yeah - that sounds familiar. If nothing else it will be interesting to run as a comparison.
I've added it back it, but renamed it to something a bit easier to type on the cmd line (`--no-client-share`)